arch/src/aarch64/mod.rs

```diff
@@ -3,11 +3,15 @@
 
 pub mod layout;
 
+use std::cmp::min;
+
 use memory_model::{GuestAddress, GuestMemory};
 
-/// Stub function that needs to be implemented when aarch64 functionality is added.
+/// Returns a Vec of the valid memory addresses for aarch64
+/// See `layout` module for a drawing of the specific memory model for this platform.
 pub fn arch_memory_regions(size: usize) -> Vec<(GuestAddress, usize)> {
-    vec![(GuestAddress(0), size)]
+    let dram_size = min(size, layout::DRAM_MEM_END);
+    vec![(GuestAddress(layout::DRAM_MEM_START), dram_size)]
```

Comment:
Judging from the ASCII art above, it seems there are multiple gaps from DRAM_START to DRAM_END, and some of them are even "Hole or DRAM". Does this warrant having multiple memory regions, instead of just one big region from DRAM_START to DRAM_END? Can setting things up with a single region lead to issues because we or the guest accidentally think we can use some memory area that's actually a gap, reserved or some other non-DRAM type of zone?

Reply:
A reserved/gap space is a space that exists but reserved for some hidden component. Reserved space does not necessarily mean that it is not available. Let s say that the firmware will need 2 MiB of RAM and then mark them as reserved. Reserved regions were really introduced to tell you that someone stole something from your DRAM. One thing that we could worry about is if the kernel entry point that we are using supports starting a kernel that is loaded at > 16 GiB but we should not have this problem since we are placing the kernel at the start of the RAM.